mono/utils/mono-threads-coop.c

```diff
@@ -16,6 +16,15 @@
 #define _DARWIN_C_SOURCE
 #endif
 
+/* On platforms that doesn't have full context support (or doesn't do conservative stack scan), use copy stack data */
+/* when entering safe/unsafe GC regions. For platforms with full context support (doing conservative stack scan), */
+/* there is already logic in place to take context before getting in a state where thread could be conservative */
```

Comment:
This is talking about `thread_state_init (&info->thread_saved_state [SELF_SUSPEND_STATE_INDEX])`?

Reply:
Yes, as part of entering safe mode we will take threads context in mono_threads_enter_gc_safe_region_unbalanced_with_info.